src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -433,8 +433,18 @@ private static bool IsConstant(IOperation operation)
 
             private bool CheckTargetExpression(IOperation operation)
             {
-                if (operation is IConversionOperation { IsImplicit: false } op)
+                if (operation is IConversionOperation op)
                 {
+                    // We can't have implicit conversions because switch arms don't do them.
+                    // Putting an explicit cast on the arm doesn't have the same semantics. For example:
+                    //    if (c == 123)
+                    // is the same as
+                    //    if ((int)c == 123)
```

Comment:
I'm confused by this:

https://sharplab.io/#v2:EYLgtghglgdgPgAQEwAIDCBYAUAb2yglBAFhQFkAKAYwAsIAnFKgSn0Ly0K5QGcB3KABdaKaq07cCHSZKoQeAUxQA+ALyjaDZgEYkADhBsZhOYpQByCOcMTjkhNoB0DgJxiA3Ebtdg9BRABrT1tuAF9CL1DsLxJyJGo6RhYvaRkoADMNFXVdPRQ4OCYUVXVLc3FvAgdnbTcWYK4orFCgA===

Reply:
No idea, I was going off debugging and seeing what operations were there. Maybe @CyrusNajmabadi can explain?

Could be a quirk of SharpLab maybe? The syntax tree doesn't show the conversion operation either, but if it wasnt there this change wouldn't work 🤷🏼‍♂️

Reply:
> The syntax tree doesn't show the conversion operation either, but if it wasnt there this change wouldn't work 🤷🏼‍♂️

The conversion exists, but I mean it always happen on the constant value side. Meaning that `if (c == 123)` is the same as `if (c == (char)123)`, which in turn is equivalent to adding the explicit cast to the switch.

Reply:
I understand what you're saying, and SharpLab shows things that way, but when debugging the conversion operation wqs definitely on the other side... Though I'm not at my PC anymore, and my confidence is rapidly waning :)

Reply:
Also if the change in semantics is a very edge (uncommon) case, we could still offer the refactoring with a warning annotation similar to:

https://github.com/dotnet/roslyn/blob/601cefd1d483cc21c43ddad48a7d683654afe8af/src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs#L208-L209

Reply:
> but when debugging the conversion operation wqs definitely on the other side

Does that mean `IOperation` is providing wrong info?